lib/rust/lexer/definition/src/lexer.rs

```diff
@@ -823,16 +1905,41 @@ impl<Logger:AnyLogger<Owned=Logger>> flexer::State for State<Logger> {
         let operator_suffix_check   = lexer_states.define_group("OPERATOR_SUFFIX_CHECK",None);
         let operator_modifier_check =
             lexer_states.define_group("OPERATOR_MODIFIER_CHECK",Some(operator_suffix_check));
-        let block_top_level     = lexer_states.define_group("BLOCK_MODULE", None);
-        let block_newline       = lexer_states.define_group("BLOCK_NEWLINE",None);
-        let in_block_line       = lexer_states.define_group("IN_BLOCK_LINE",None);
-        let tokens_stack        = Vec::new();
-        let offset_logger       = <Logger>::sub(&logger,"Offset");
-        let offset              = Offset::new(offset_logger);
-        let number_state_logger = <Logger>::sub(&logger,"NumberState");
-        let number_state        = NumberLexingState::new(number_state_logger);
-        let block_state_logger  = <Logger>::sub(&logger,"BlockLexingState");
-        let block_state         = BlockLexingState::new(block_state_logger);
+        let block_top_level  = lexer_states.define_group("BLOCK_MODULE", None);
+        let block_newline    = lexer_states.define_group("BLOCK_NEWLINE",None);
+        let in_block_line    = lexer_states.define_group("IN_BLOCK_LINE",None);
+        let text_escape      = lexer_states.define_group("TEXT_ESCAPE",None);
+        let text_format      = lexer_states.define_group("TEXT_FORMAT",Some(text_escape));
+        let text_format_line = lexer_states.define_group("TEXT_FORMAT_LINE",Some(text_format));
+        let text_format_inline_block =
+            lexer_states.define_group("TEXT_FORMAT_INLINE_BLOCK",Some(text_format));
+        let text_format_block = lexer_states.define_group("TEXT_FORMAT_BLOCK",Some(text_format));
+        let text_format_line_nested =
+            lexer_states.define_group("TEXT_FORMAT_LINE_NESTED",Some(text_format_line));
+        let text_raw      = lexer_states.define_group("TEXT_RAW",None);
+        let text_raw_line = lexer_states.define_group("TEXT_RAW_LINE",Some(text_raw));
+        let text_raw_inline_block =
+            lexer_states.define_group("TEXT_RAW_INLINE_BLOCK",Some(text_raw));
+        let text_raw_block = lexer_states.define_group("TEXT_RAW_BLOCK",Some(text_raw));
+        let text_raw_line_nested =
+            lexer_states.define_group("TEXT_RAW_LINE_NESTED",Some(text_raw_line));
+        let text_seen_newline    = lexer_states.define_group("TEXT_SEEN_NEWLINE",None);
+        let text_interpolate     = lexer_states.define_group("TEXT_INTERPOLATE",Some(initial_state));
+        let comment              = lexer_states.define_group("COMMENT",None);
+        let disable_comment      = lexer_states.define_group("DISABLE_COMMENT",Some(comment));
+        let doc_comment          = lexer_states.define_group("DOC_COMMENT",Some(comment));
+        let doc_comment_newline  = lexer_states.define_group("DOC_COMMENT_NEWLINE",None);
+        let tokens_stack         = Vec::new();
```

Comment:
lets talk how to automate these, as keeping it up to date is error prone and requires magical knowledge